src/main/java/tc/oc/pgm/modules/InfoModule.java

```diff
@@ -117,6 +129,22 @@ public static InfoModule parse(MapModuleContext context, Logger logger, Document
           throw new InvalidXMLException("Contributor must have either a name or UUID", child);
         }
 
+        if (uuid != null) {
+          if (Contributor.isUUIDCached(uuid)) {
+            name = Contributor.getCachedName(uuid);
+          } else {
+            Bukkit.getScheduler()
+                .runTaskAsynchronously(
```

Comment:
This is creating a new bukkit scheduled task for every contributor UUID, and if there are any repeated ones you may be querying for them multiple times.

Consider doing this once (instead of per-map & per-author) with all the unique different uuids loaded by all maps